transform/convert_quadkey_and_Vertical_id.go

```diff
@@ -576,6 +662,122 @@ func convertVerticallIDToBit(vZoom int64, vIndex int64, outputZoom int64, maxHei
 
 }
 
+// transforms the vertical index with a given index-ZoomLevel pair and returns the vertical index(es) that occupy the same vertical space with a new zoomLevel, zoomScalar, and vertical offset.
+//
+// input:
+//
+//	inputIndex: the index to transform
+//
+//	inputZoom: the zoomLevel of the inputIndex
+//
+//	outputZoom: the zoomLevel of the outputIndex, before using zoomScalar (net outputZoom = outputZoom+zoomScalar)
+//
+//	zoomScalar: a scalar to add to outputZoom (net outputZoom = outputZoom+zoomScalar)
+//
+//	offset: an integer to shift the height of the inputIndex up or down. A positive offset means the transformed index is higher in altitude than that of the input.
+//
+// output:
+//
+//	[]int64 list of vertical index(es), error
+func convertVerticalIndex(inputIndex int64, inputZoom int64, outputZoom int64, altitudeRangeScalar int64, indexOffset int64) ([]int64, error) {
```

Comment:
Rename altitudeRangeScalar to zoomOffset.
If you remove outputZoom like the following one, the return value will no longer be a slice and will be very simple.
Don't you remove outputZoom?
```
func convertVerticalIndex(inputIndex int64, zoomOffset int64, indexOffset int64) int64 {
    return inputIndex >> zoomOffset + indexOffset
}
```

Reply:
altitudeRangeScalar scales the altitudeRange by a factor of 2^n, whereas indexOffset shifts the altitude range up or down linearly

Reply:
altitudeRange of SpatialIDs is 33554432 - -33554432 = 2^26, right?
So altitudeRange of QuadkeysAndAltitudekeys is 15872 - -512=2^14.
But you set 26 - 14 = 12 ≒ 11 as altitudeRangeScalar.
I think this is not altitudeRangeScalar but zoomOffset.

Reply:
MEMO: @buscandoaverroes rejects the proposal by @HarutakaMatsumoto.